### PR TITLE
chore(examples) Use MapboxOverlay in mapbox (formerly safegraph) website example

### PR DESCRIPTION
**Pull request by chrisgervang**
For #8557

#### Change List
- Replace MapboxLayer with MapboxOverlay
- Rename website example to `mapbox`
- Use react instead of pure-js

### CODE REVIEW

File: examples/website/safegraph/app.js [hunk 1]
```diff
@@ -79,14 +84,18 @@ function renderLayers(map, data) {
...
-      data: data.filter(d => d.hex === hex)
-    });
+    deckOverlay.setProps({ 
+      layers: [
+        poiLayer, // TODO: This was written with an imperative pattern, which won't work for MapboxOverlay. poiLayer needs to be defined.
+        arcLayer.clone({
+          data: data.filter(d => d.hex === hex)
+        })
+      ]
+    })
```
Comment (chrisgervang) [inline examples/website/safegraph/app.js:94]: I'd like to hear some opinions on how to demonstrate this kind of partial update in an example. `MapboxLayer` supported it with a special `setProps` function added to each layer, but this is inconsistent with the majority of our examples and isn't supported when regular layers are in use, as is the case when using `MapboxOverlay`.

Comment (felixpalmer) [reply examples/website/safegraph/app.js:94]: The same behavior can be done using the `MaskExtension`: https://github.com/visgl/deck.gl/blob/master/test/apps/mask/app.jsx

If not using that, I think it is OK to recreate all the layers for the `setProps` call, deck.gl should do the diff internally and only update what is needed

File: examples/website/safegraph/app.js [hunk 2]
```diff
@@ -48,26 +52,27 @@ export function renderToDOM(container, data) {
...
-    update: newData => renderLayers(map, newData),
+    update: newData => renderLayers(map, deckOverlay, newData),
     remove: () => {
       map.remove();
     }
   };
 }
 
-function renderLayers(map, data) {
+function renderLayers(map, deckOverlay, data) {
```
Comment (Pessimistress) [inline examples/website/safegraph/app.js:68]: Is `map` still needed in this function?

File: examples/website/safegraph/app.js (hunk 1 above)
Comment (Pessimistress) [reply examples/website/safegraph/app.js:94]: The common way of doing this is to change the signature to `renderLayers(deckOverlay, data, selectedPOI)` and call it from `onClick`.

File: examples/website/safegraph/app.js (hunk 2 above)
Comment (chrisgervang) [reply examples/website/safegraph/app.js:68]: It's used for `beforeId: getFirstLabelLayerId(map.getStyle())`

File: examples/website/safegraph/app.js (hunk 1 above)
Comment (chrisgervang) [reply examples/website/safegraph/app.js]: Thanks! Got it to work.
mapbox-v9

File: examples/website/safegraph/app.js [hunk 3]
```diff
@@ -48,60 +53,51 @@ export function renderToDOM(container, data) {
...
-    update: newData => renderLayers(map, newData),
+    update: newData => renderLayers(map, deckOverlay, newData),
     remove: () => {
       map.remove();
     }
   };
 }
 
-function renderLayers(map, data) {
+function renderLayers(map, deckOverlay, data, selectedPOI) {
```
[the remark this answers is not in the document]
Comment (chrisgervang) [reply examples/website/safegraph/app.js:69]: It's used for `beforeId`

[the remark this answers is not in the document]
Comment (Pessimistress) [reply examples/website/safegraph/app.js:69]: I'm not a fan of this function. It loops in too many things that are not relevant to the reason of rerender (selecting a hex).

**Comment by Pessimistress** [comment]
I thought we were renaming this example to `mapbox`?

**Comment by chrisgervang** [comment]
@Pessimistress, as discussed, I've switched the example to use react. Lmk if you have any comments or if this is ready to merge.

**Review by Pessimistress — APPROVED** [review]

**Pull request merged**